src/store/modules/wallet.js

```diff
@@ -563,11 +566,59 @@ const actions = {
       return true;
     });
 
-    if (filteredEvents?.length) {
-      const existingEvents = state.followeeEvents || [];
-      const updatedEvents = [...existingEvents, ...filteredEvents];
-      commit(WALLET_SET_FOLLOWEE_EVENTS, updatedEvents);
+    // 4. group batch send events
+    const batchSendEventsMap = new Map();
+
+    const isPotentialBatchSendEvent = ({ type, sender, txHash }) =>
+      type === 'send' &&
+      followees.includes(sender) &&
+      !batchSendEventsMap.has(txHash);
+
+    const isBatchSendEvent = ({ type, sender, txHash }) =>
+      type === 'send' &&
+      followees.includes(sender) &&
+      batchSendEventsMap.has(txHash);
+
+    filteredEvents.forEach(event => {
+      const { type, sender, tx_hash: txHash, receiver } = event;
+      if (isPotentialBatchSendEvent({ type, sender, txHash })) {
+        batchSendEventsMap.set(txHash, { ...event, batchSendList: [receiver] });
+      } else if (isBatchSendEvent({ type, sender, txHash })) {
+        const storedBatchSendEvent = batchSendEventsMap.get(txHash);
+        if (!storedBatchSendEvent.batchSendList.includes(receiver)) {
+          storedBatchSendEvent.batchSendList.push(receiver);
+        }
+      }
+      if (filteredEvents?.length) {
+        filteredEvents.forEach(event => {
+          const existingEvents = state.followeeEventsMap[event.tx_hash];
+          commit(WALLET_SET_FOLLOWEE_EVENTS, {
+            txHash: event.tx_hash,
+            event: { ...existingEvents, ...event },
+          });
```

Comment:
We can check for `event.tx_hash` in `batchSendEventsMap` here and commit batch send here, instead of looping again in below block

Also we might want to handle cases where `send` and other type of event are mixed under same `tx_hash`, we should still commit non-send event event if we commited batch send

Maybe we should commit batch send version first, then commit other non-batch or non-send events?